### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by sornas**
Part of #31

### CODE REVIEW

**Comment by sornas** [comment]
Honestly, we should merge these now and work on one part at a time. I'll update to match #102 and then mark as ready.

File: mumlib/src/state.rs [hunk 1]
```diff
@@ -12,14 +12,33 @@ pub struct Server {
 #[derive(Clone, Debug, Deserialize, Serialize)]
 pub struct Channel {
```
Comment (default-username-852) [inline mumlib/src/state.rs:13]: Document this struct and the Server struct.

File: mumlib/src/state.rs [hunk 2]
```diff
@@ -12,14 +12,33 @@ pub struct Server {
...
     pub max_users: u32,
     pub name: String,
     pub children: Vec<Channel>,
     pub users: Vec<User>,
+
+    links: Vec<Vec<usize>>, //to represent several walks through the tree to find channels its linked to
 }
 
 impl Channel {
+    pub fn new(
```
Comment (default-username-852) [inline mumlib/src/state.rs:24]: Document this function.

File: mumlib/src/state.rs [hunk 3]
```diff
@@ -12,14 +12,33 @@ pub struct Server {
...
+            max_users,
+            name,
+            children: Vec::new(),
+            users: Vec::new(),
+
+            links: Vec::new(),
+        }
+    }
+
+    /// Create an iterator over this channel and its children in a pre-order
```
Comment (default-username-852) [inline mumlib/src/state.rs:40]: What is a pre-order traversal?

**Review by default-username-852 — CHANGES_REQUESTED** [review]
Great job in general! It might be nice to add some module level documentation to mumlib as well, and maybe add a `#![warn(missind_docs)]`, if the aim of this PR is to document all of mumlib.

**Comment by sornas** [comment]
> and maybe add a `#![warn(missind_docs)]`

Still 85 missing documentations in mumlib, so no. Not yet.

**Review by default-username-852 — APPROVED** [review]
Looks good!

**Pull request merged**